src/OpenTelemetry/Metrics/Aggregator/LastValueAggregator.cs

```diff
@@ -0,0 +1,75 @@
+// <copyright file="LastValueAggregator.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Metrics;
+using System.Linq;
+
+namespace OpenTelemetry.Metrics
+{
+    public class LastValueAggregator : Aggregator
+    {
+        private readonly Instrument instrument;
+        private readonly string[] names;
+        private readonly object[] values;
+        private IDataPoint lastValue = null;
+        private int count = 0;
+
+        public LastValueAggregator(Instrument instrument, string[] names, object[] values)
+        {
+            this.instrument = instrument;
+            this.names = names;
+            this.values = values;
+        }
+
+        public override void Update(IDataPoint value)
+        {
+            this.count++;
+            this.lastValue = value;
+        }
+
+        public override IEnumerable<Metric> Collect()
+        {
+            // TODO: Need to determine how to convert to Metric
+
+            if (this.count == 0)
+            {
+                return Enumerable.Empty<Metric>();
+            }
+
+            var attribs = new List<KeyValuePair<string, object>>();
+            for (int i = 0; i < this.names.Length; i++)
+            {
+                attribs.Add(new KeyValuePair<string, object>(this.names[i], this.values[i]));
+            }
+
+            var dp = this.lastValue.NewWithValue();
```

Comment:
This line is a bit confusing. Is `NewWithValue` kind of like a `Clone`?

Reply:
Yeah, the name is bad. I'll rename to Clone(KeyValuePair<string, object>[] tags) unless you have a better suggestion.

Also, I'm looking for a better approach here also. The issue is how to make a new DataPoint<T> without needing to deal with T or testing against T.